implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxMeterRegistry.java

```diff
@@ -118,6 +129,22 @@ protected void publish() {
         }
     }
 
+    @Override
+    protected Timer newTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig, PauseDetector pauseDetector) {
+        // TODO: Investigate if the default HistogramGauges metrics should be removed:
+        //  * Adds a new gauge ".histogram" with "le" tag when slo buckets defined.
+        //  * Adds a new gauge ".percentile" with "phi" tag when percentiles buckets are defined.
+        return super.newTimer(id, updateConfig(distributionStatisticConfig), pauseDetector);
+    }
+
+    @Override
+    protected DistributionSummary newDistributionSummary(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig, double scale) {
+        // TODO: Investigate if the default HistogramGauges metrics should be removed:
+        //  * Adds a new gauge ".histogram" with "le" tag when slo buckets defined.
```

Comment:
The `.histogram` gauge seems to be almost equivalent to what you've implemented here, aside from the naming (`_bucket` meter name, `upper_bound` vs `le` tag) and the additional SLO bucket.
Have you considered using `NamingConvention` & `MeterFilter` to modify the configuration of the default `HistogramGauges`-generated meters?

Reply:
Thanks for pointing to that (I looked into it before, but tried one more time):
* The upper_bound is as simple as asking the customer to add it manually which is not ideal but ok.
* The biggest problem is the fact that the reported metrics are Gauges not CumulativeCounters. I can do a small hack on our side to rewrite the type to be Counter.